google-cloud-clients/google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java

```diff
@@ -607,6 +608,35 @@ public void testListTables() {
     assertTrue(createdTable.delete());
   }
 
+  @Test
+  public void testListTablesWithPartitioning() {
+    String tableName = "test_list_tables_partitioning";
+    TimePartitioning timePartitioning = TimePartitioning.of(Type.DAY, EXPIRATION_MS);
+    StandardTableDefinition tableDefinition =
+        StandardTableDefinition.newBuilder()
+            .setSchema(TABLE_SCHEMA)
+            .setTimePartitioning(timePartitioning)
+            .build();
+    TableInfo tableInfo = TableInfo.of(TableId.of(DATASET, tableName), tableDefinition);
+    Table createdPartitioningTable = bigquery.create(tableInfo);
+    assertNotNull(createdPartitioningTable);
+    Page<Table> tables = bigquery.listTables(DATASET);
+    boolean found = false;
+    Iterator<Table> tableIterator = tables.getValues().iterator();
+    while (tableIterator.hasNext() && !found) {
+      StandardTableDefinition standardTableDefinition = tableIterator.next().getDefinition();
+      if (standardTableDefinition.getTimePartitioning().getType().equals(Type.DAY)
+          && standardTableDefinition
+              .getTimePartitioning()
+              .getExpirationMs()
+              .equals(EXPIRATION_MS)) {
+        found = true;
+      }
+    }
+    assertTrue(found);
+    assertTrue(createdPartitioningTable.delete());
```

Comment:
If delete always needs to happen, just do it in a finally clause without an assert..

Reply:
Done